rules/the-difference-between-data-transfer-objects-and-view-models/rule.md

```diff
@@ -8,19 +8,77 @@ created: 2019-04-15T21:58:44.0000000Z
 authors:
 - title: Jason Taylor
   url: https://ssw.com.au/people/jason-taylor
-related: []
+- title: Matt Goldman
+  url: https://ssw.com.au/people/matt-goldman
+related:
+  - use-mvvm-pattern
 redirects:
 - do-you-know-the-difference-between-data-transfer-objects-and-view-models
 
 ---
 
-Data Transfer Objects (DTOs) and View Models (VMs) are not the same concept! The main difference is that while VMs can encapsulate behaviour, DTOs do not.
+Data Transfer Objects (DTOs) and View Models (VMs) are similar concepts, but the terms are not interchangeable. To make matters more confusing the term `ViewModel` has a special meaning inside the [MVVM pattern](/use-mvvm-pattern).
 
-The purpose of a DTO is the transfer of data from one part of an application to another. Since DTOs do not encapsulate behaviour, they can easily be serialised and deserialized into other formats, e.g. JSON, XML, and so on.
+Do you understand the subtle difference between these terms?
 
 <!--endintro-->
 
-The purpose of a VM is also the transfer of data, however VMs can encapsulate behaviour. This behaviour is useful, for example, when creating a WPF + MVVM application, but not so useful when creating a SPA - since you can't serialize the behaviour and pass it from ASP.NET MVC to the client.
+## What is a DTO?
+
+A DTO is a type that is used to move data from one part of an application to another. This could be between a command and a service, or between an API and a UI. Typically, a DTO represents an entity or other resource.
+
+Unlike regular classes, a DTO only includes data - not behaviour.
+
+```csharp
+public class CarDto
+{
+  public int Id { get; set; }
+
+  public int MakeId { get; set; }
+  public string Make { get; set; }
+  
+  public int ModelId { get; set; }
+  public string Model { get; set; }
+
+  public int Year { get; set; }
+}
+```
+::: good
+Good example - A DTO that encapsulates some data
```

Comment:
we usually show the BAD then the GOOD :)

Reply:
Done